core/domain/html_cleaner.py

```diff
@@ -509,6 +509,18 @@ def convert_to_ckeditor(html_data):
         while li.parent.name in ['li', 'p']:
             li.parent.unwrap()
 
+    # Ensure li is wrapped in ol/ul.
+    for li in soup.findAll('li'):
+        if li.parent.name not in ['ol', 'ul']:
+            new_parent = soup.new_tag('ul')
+            next_sib = list(li.next_siblings)
```

Comment:
Why not just do `for sib in li.next_siblings:`? Doesn't look like `next_sib` is used for anything else.

Or if you want to keep `next_sib`, do it right before the start of the `for` loop (ie, swap the order of this line and the next line for readability)

Reply:
I need to extract the next siblings before wrapping li tag in new parent otherwise the siblings will change since now the tag has a different parent. So, we need to keep the same order.

Reply:
Ah got it, thanks for explaining.